### DISCUSSION TITLE
Configurable triangulation of HeightFieldShape

### DISCUSSION THREAD

**Discussion by mihe** [jrouwe/JoltPhysics#680] [Q&A] [answered]
It was brought to my attention ([here](https://github.com/godot-jolt/godot-jolt/issues/566)) that I'd made a somewhat significant oversight when implementing Godot's `HeightMapShape3D`, which is that the way the shape is triangulated in Jolt differs from how it's triangulated in Godot Physics, where the diagonal edge runs from (what you refer to as) `x2y1` to `x1y2` rather than `x2y2` to `x1y1`.

For height maps with a high enough resolution I guess this doesn't really matter all that much, seeing as how this has gone unreported until now, but for lower resolutions with sharp crevices or steep inclines that run diagonally it ends up making quite a big difference.

Looking through the code I get the sense that this would not be a trivial thing to make configurable without making things a lot messier, but I figured I would check with you anyway to see what could be done, if anything. If you can see some hacky way of shuffling things around that would resolve this, but that might not be desirable (or even make sense) for mainline Jolt, I'd be perfectly fine with having my fork diverge as well.

**Comment by jrouwe** [comment, marked as answer]
Isn't this as easy as wrapping the shape in a ScaledShape with scale (1 1 -1) and reversing the height field data rows? (Or rotating by 90 degrees and flipping the height field data)

**Reply by mihe** [reply]
Yes, it is apparently, and I feel dumb. :)

Thank you!
